install

```diff
@@ -13,69 +19,125 @@ usage() {
-        "$CURRENT_DIR/uninstall"
-      fi
-
-      if [[ $? != 0 ]]; then
-        echoc "There was a problem uninstalling the previous fastlane bundle, aborting installation..." red
-        exit 1
-      fi
-      echoc "Done uninstalling, your regular installation will now commence. 🚕 💨" green
-      ;;
-    * )
-      echoc "Installation aborted, nothing to be done!" green
-      exit 1
-      ;;
+prompt_to_install_or_upgrade()
+{
+  PACKAGE=$1
+
+  read -p "The installer needs to install/upgrade $PACKAGE with brew. Is that okay? (y/N) " -n 1 choice
+  echo ""
+
+  case "$choice" in
+  y|Y )
+    # Nothing
+  ;;
+  * )
+    echoc "Cancelled uninstall process" red
+    exit 1
+  ;;
   esac
+}
+
+# Check if fastlane already exists
+if [ -d "$FASTLANE_DIR" ]; then
+  echoc "fastlane is already installed at $FASTLANE_DIR" yellow
+
+  # Check if old version and clean it up
+  if [ ! -f $PACKAGED_FASTLANE_VERSION_FILE ]; then
+    echoc "Found old version of brew packaged fastlane. Cleaning up..." yellow
+    "$CURRENT_DIR/uninstall" -y
+  fi
 fi
 
-# Copy fastlane to ~/.fastlane
-echoc "Installing fastlane to $FASTLANE_DIR... this might take a few seconds" yellow
-mkdir -p $FASTLANE_DIR
-# We have to skip the 2 error messages below, which are shown if a previous version of fastlane
-# was installed via the bundle
-cp -R "$CURRENT_DIR/fastlane_lib/" $FASTLANE_DIR
+echoc "Installing fastlane to $FASTLANE_DIR..." yellow
+
+#
+# Install ruby-build and Ruby version $PACKAGED_RUBY_VERSION only if needed
+#
+if [[ -f $PACKAGED_RUBY_VERSION_FILE && $(< $PACKAGED_RUBY_VERSION_FILE) == "$PACKAGED_RUBY_VERSION" ]]; then
+  echo "Ruby $PACKAGED_RUBY_VERSION already installed"
+else
+  echo "Ruby $PACKAGED_RUBY_VERSION not installed. Preparing to install now. This might take a few minutes"
+  brew update >/dev/null 2>&1
+
+  install_or_upgrade_via_homebrew "ruby-build"
+  install_or_upgrade_via_homebrew "openssl"
+
+  echoc "Building Ruby $PACKAGED_RUBY_VERSION... this also might take a few minutes" yellow
+
+  mkdir -p "$FASTLANE_DIR/bundle"
+
+  OPENSSL_DIR=$(brew --prefix openssl)
+  if [ $? -ne 0 ]; then
+    echoc "Couldn't find path of OpenSSL. Cancelled install process..." red
+    exit 1
+  fi
+
+  echo "Linking Ruby $PACKAGED_RUBY_VERSION to OpenSSL at $OPENSSL_DIR"
+  RUBY_CONFIGURE_OPTS=--with-openssl-dir="$OPENSSL_DIR" ruby-build $PACKAGED_RUBY_VERSION "$FASTLANE_DIR/bundle"
+
+  echo "$PACKAGED_RUBY_VERSION" >> $PACKAGED_RUBY_VERSION_FILE
+fi
+
+# Copy helpers for keeping a clean execution environment
+cp "$CURRENT_DIR/bundle-env" "$FASTLANE_DIR/bundle/bin"
+cp "$CURRENT_DIR/parse_env.rb" "$FASTLANE_DIR"
+
+# Install bundler and fastlane
+$FASTLANE_DIR/bundle/bin/bundle-env gem install bundler --no-document --env-shebang
+$FASTLANE_DIR/bundle/bin/bundle-env gem install fastlane --no-document --env-shebang
+
+# Copy the fastlane executable to run fastlane in the bundled environment
+cp "$CURRENT_DIR/fastlane_shim" "$FASTLANE_DIR/fastlane"
```

Comment:
You deleted this file.

Reply:
Ahhh, that was supposed to be committed! Thanks 😊